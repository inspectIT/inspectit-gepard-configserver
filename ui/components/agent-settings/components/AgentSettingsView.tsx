import { Connection } from "@/components/connections/interfaces/Connection";
import AgentDetails from "./AgentDetails";
import AgentScopeForm from "./AgentScopeForm";

interface AgentSettingsViewProps {
  connection: Connection;
}

/*
Renders the agent settings view with the agent details and the agent scope form.
*/
export default function AgentSettingsView({
  connection,
}: AgentSettingsViewProps) {
  const startTime = connection.startTime;
  const date = new Date(startTime);

  return (
    <div className="flex flex-col gap-2">
      <AgentDetails connection={connection} />
      <AgentScopeForm connectionId={connection.id} />
    </div>
  );
}
