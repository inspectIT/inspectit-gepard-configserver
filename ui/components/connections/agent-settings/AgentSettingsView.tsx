import { Agent } from "../Agent";
import AgentDetails from "./AgentDetails";
import AgentScopeForm from "./AgentScopeForm";

interface AgentSettingsViewProps {
  agent: Agent;
}

export default function AgentSettingsView({ agent }: AgentSettingsViewProps) {
  const startTime = agent.startTime;
  const date = new Date(startTime);

  return (
    <div className="flex flex-col gap-2">
      <AgentDetails agent={agent} />
      <AgentScopeForm agentId={agent.id} />
    </div>
  );
}
