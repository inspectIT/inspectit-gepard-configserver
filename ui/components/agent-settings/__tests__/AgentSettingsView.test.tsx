import { describe, expect, test } from "vitest";
import { render } from "@testing-library/react";
import AgentSettingsView from "../components/AgentSettingsView";
import { TestConnectionsSmall } from "@/components/connections/__tests__/TestConnections";

describe("AgentSettings Page", () => {
  test("matches snapshot", () => {
    const rerender = render(
      <AgentSettingsView connection={TestConnectionsSmall.at(0)!} />
    );
    expect(rerender).toMatchSnapshot();
  });
});
