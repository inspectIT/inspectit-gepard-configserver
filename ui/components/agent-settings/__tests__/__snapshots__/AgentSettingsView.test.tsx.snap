// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AgentSettings Page > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col gap-2"
      >
        <div
          class="rounded-lg border bg-card text-card-foreground shadow-sm p-2"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <h3
              class="text-2xl font-semibold leading-none tracking-tight"
            >
              Agent
            </h3>
            <p
              class="text-sm text-muted-foreground"
            >
              Running on 
              1
            </p>
          </div>
          <div
            class="p-6 pt-0"
          >
            <div
              class="space-y-1"
            >
              <h4
                class="text-sm leading-none"
              >
                InspectIT Gepard Version
              </h4>
              <p
                class="text-sm text-muted-foreground"
              >
                0.0.1-SNAPSHOT
              </p>
            </div>
            <div
              class="space-y-1"
            >
              <h4
                class="text-sm leading-none"
              >
                OpenTelemetry Version
              </h4>
              <p
                class="text-sm text-muted-foreground"
              >
                0.0.1-SNAPSHOT
              </p>
            </div>
            <div
              class="space-y-1"
            >
              <h4
                class="text-sm leading-none"
              >
                Java Version
              </h4>
              <p
                class="text-sm text-muted-foreground"
              >
                17
              </p>
            </div>
          </div>
        </div>
        <div
          class="rounded-lg border bg-card text-card-foreground shadow-sm"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <h3
              class="text-2xl font-semibold leading-none tracking-tight"
            >
              Scopes
            </h3>
            <p
              class="text-sm text-muted-foreground"
            >
              Configure the scope of this agent.
            </p>
          </div>
          <div
            class="p-6 pt-0"
          >
            <h1
              class="text-2xl font-bold"
            >
              Here you will have an Input Field and a List of Classes, which this agents configures.
            </h1>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col gap-2"
    >
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm p-2"
      >
        <div
          class="flex flex-col space-y-1.5 p-6"
        >
          <h3
            class="text-2xl font-semibold leading-none tracking-tight"
          >
            Agent
          </h3>
          <p
            class="text-sm text-muted-foreground"
          >
            Running on 
            1
          </p>
        </div>
        <div
          class="p-6 pt-0"
        >
          <div
            class="space-y-1"
          >
            <h4
              class="text-sm leading-none"
            >
              InspectIT Gepard Version
            </h4>
            <p
              class="text-sm text-muted-foreground"
            >
              0.0.1-SNAPSHOT
            </p>
          </div>
          <div
            class="space-y-1"
          >
            <h4
              class="text-sm leading-none"
            >
              OpenTelemetry Version
            </h4>
            <p
              class="text-sm text-muted-foreground"
            >
              0.0.1-SNAPSHOT
            </p>
          </div>
          <div
            class="space-y-1"
          >
            <h4
              class="text-sm leading-none"
            >
              Java Version
            </h4>
            <p
              class="text-sm text-muted-foreground"
            >
              17
            </p>
          </div>
        </div>
      </div>
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm"
      >
        <div
          class="flex flex-col space-y-1.5 p-6"
        >
          <h3
            class="text-2xl font-semibold leading-none tracking-tight"
          >
            Scopes
          </h3>
          <p
            class="text-sm text-muted-foreground"
          >
            Configure the scope of this agent.
          </p>
        </div>
        <div
          class="p-6 pt-0"
        >
          <h1
            class="text-2xl font-bold"
          >
            Here you will have an Input Field and a List of Classes, which this agents configures.
          </h1>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
